app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/MongoClientCustomizer.java

```diff
@@ -58,18 +68,28 @@ public void customize(ComponentProxyComponent component, Map<String, Object> opt
                     MongoClientURI mongoClientURI = new MongoClientURI(mongoConf.getMongoClientURI());
                     MongoClient mongoClient = new MongoClient(mongoClientURI);
                     options.put("mongoConnection", mongoClient);
-                    if(!options.containsKey("connectionBean")) {
+                    if (!options.containsKey("connectionBean")) {
                         //We safely put a default name instead of leaving null
-                        options.put("connectionBean", String.format("%s-%s",mongoConf.getHost(),mongoConf.getUser()));
+                        options.put("connectionBean", String.format("%s-%s", mongoConf.getHost(), mongoConf.getUser()));
                     }
                 } catch (@SuppressWarnings("PMD.AvoidCatchingGenericException") Exception e) {
                     throw new IllegalArgumentException(e);
                 }
             } else {
                 LOGGER.warn(
-                        "Not enough information provided to set-up the MongoDB client. Required host, user and password.");
+                    "Not enough information provided to set-up the MongoDB client. Required at least host, user and password.");
             }
         }
     }
+
+    // Validate the operation among a list of possible values
+    @SuppressWarnings("PMD")
```

Comment:
😢 can we find code that is valid according to PMD? At least check for null value to avoid the NPE? And the MongoProducerOperation could provide a isValid helper providing the formatted IllegalArgumentException

Reply:
I don't like that either, I fell in the temptation to simply put a suppress warning as passing a non valid enumeration just throw the same exception... gonna refactor that!